src/comps/temp-withdraw.tsx

```diff
@@ -0,0 +1,240 @@
+    placeholder: "the max fee",
+  },
+];
+
+const BasicWithdraw = () => {
+  const [txId, setTxId] = useState<string | null>(null);
+  const { addresses, selectedWallet } = useAtomValue(walletInfoAtom);
+
+  const router = useRouter();
+
+  const { WALLET_NETWORK, SBTC_CONTRACT_DEPLOYER } =
+    useAtomValue(bridgeConfigAtom);
+
+  const handleSubmit = async (values: Record<string, string>) => {
+    console.log(values);
+    const { address, amount, fee } = values;
+
+    if (!WALLET_NETWORK) {
+      throw new Error("Invalid network");
+    }
+    if (!SBTC_CONTRACT_DEPLOYER) {
+      throw new Error("Invalid contract deployer");
+    }
+    // get the proper data for the address
+    const recipient = decodeBitcoinAddressToClarityRecipient(
+      address,
+      WALLET_NETWORK,
+    );
+
+    if (!recipient) {
+      throw new Error("Invalid recipient address");
+    }
+    const publicKey = addresses.stacks?.publicKey;
+
+    if (!publicKey) {
+      throw new Error("Invalid public key");
+    }
+    console.log(recipient);
+
+    // convert the amount to satoshis
+    const satoshiAmount = Math.round(parseFloat(amount) * 1e8);
+    console.log("satoshiAmount", satoshiAmount);
+
+    // convert the fee to satoshis
+    const satoshiFee = Math.round(parseFloat(fee) * 1e8);
+
+    const contractArgs = [
+      uintCV(satoshiAmount),
+      tupleCV({
+        version: bufferCV(hexToBytes(recipient.version)),
+        hashbytes: bufferCV(hexToBytes(recipient.hashbytes)),
+      }),
+      uintCV(satoshiFee),
+    ];
+
+    const stacksNetwork = getStacksNetwork(WALLET_NETWORK);
+
+    const transaction = await makeUnsignedContractCall({
+      contractAddress: SBTC_CONTRACT_DEPLOYER,
+      contractName: "sbtc-withdrawal",
+      functionName: "initiate-withdrawal-request",
+      functionArgs: contractArgs,
+      publicKey,
+      validateWithAbi: true,
+      network: stacksNetwork,
+      fee: WALLET_NETWORK === "mainnet" ? undefined : 10_000,
+      postConditionMode: PostConditionMode.Allow,
+    });
+
+    const signTx = {
+      [WalletProvider.XVERSE]: callContractXverse,
+      [WalletProvider.LEATHER]: callContractLeather,
+      [WalletProvider.FORDEFI]: callContractFordefi,
+      [WalletProvider.ASIGNA]: callContractAsigna,
+    }[selectedWallet!];
+
+    const signedTx = await signTx({
+      txHex: serializeTransaction(transaction),
+      network: stacksNetwork,
+    });
+
+    const broadcastResponse = await serverBroadcastTx({
+      txHex: signedTx,
+    });
+    const txId = broadcastResponse.txid;
+    router.push(`/withdraw?txId=${txId}`);
+    // add tx to search query for the user to have a link to the tx
+
+    setTxId(txId);
+  };
+  return (
+    <FlowContainer>
+      <>
+        <div className="w-full flex flex-row items-center justify-between">
+          <Heading>Start Withdraw Transfer</Heading>
+        </div>
+        <div className="flex flex-col gap-1">
+          <SubText>Sender Address</SubText>
```

Comment:
nit: TxID or something, also can we hide this if txId is null?

Reply:
nit: it's a stacks tx

Reply:
my bad, fixed